### ISSUE TITLE
PATH not being inherited correctly in Yosemite

### ISSUE THREAD

**Issue by steven-murray** [atom/atom#8104]
Hi,

I have the newest version of atom installed (1.0.2), on OSX Yosemite. I have the shell scripts installed, but when starting atom from command line (iTerm2), the PATH variable is inherited wrongly. This is screwing up atom-script when trying to run my python scripts (I'm using Anaconda python distribution which is not being picked up inside Atom) and also means it can't find pep8 and all that stuff. 

So when I use `echo $PATH` in iTerm2, I get 

```
opt/local/bin:/opt/local/sbin:/usr/local/Cellar/ruby/1.9.3-p194/bin:/Users/Steven/anaconda/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/Library/TeX/texbin
```

However, when inside atom, using the terminal package, I call the same thing `echo $PATH` and get the following (sorry couldn't copy and paste?!):

![atom_screenshot](https://cloud.githubusercontent.com/assets/1272030/8923322/c3a3ea5c-3521-11e5-84a8-d4df3f1ec5f9.png)

This matches the value of `process.env.PATH` as shown in the console. 

Where did it even get the `mrmpi-11Mar13` value from?

My `.bash_profile` exports look like this:

```
export PATH="/usr/local/bin:$PATH"

# added by Anaconda 2.2.0 installer
export PATH="/Users/Steven/anaconda/bin:$PATH"

#Change to homebrew ruby
export PATH="/usr/local/Cellar/ruby/1.9.3-p194/bin:$PATH"

# MacPorts Installer addition on 2012-06-05_at_12:55:53: adding an appropriate PATH variable for use with MacPorts.
export PATH="/opt/local/bin:/opt/local/sbin:$PATH"
```

I've tried uninstalling and re-installing atom to no avail.

**Comment by mnquintana** [comment]
This seems like a duplicate of https://github.com/atom/atom/issues/6956 – closing in favor of that issue. Let me know if I'm misunderstanding though.

**Comment by steven-murray** [comment]
No, I'm starting from the terminal, so not quite the same issue. I've just realised that there were some entries to the PATH being set from a long-forgotten ~/.profile file. However, having moved that out, I still get `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:` at the front of my PATH in atom, after which comes the stuff that I actually set in my .bash_profile.

Any idea where those could be coming from? Are they just there by default in atom? Because they're screwing up my path to python!

**Comment by mnquintana** [comment]
@steven-murray Can you try quitting Atom completely first, then checking `process.env.PATH` in the console? Also, can you try checking `process.env.PATH` in both the terminal package and the Dev Tools Console? (<kbd>Cmd-Opt-I</kbd>)

**Comment by steven-murray** [comment]
Let me do this systematically. I just tried to reproduce it with the following exact steps:

1. Complete quite Atom.
2. Open a new terminal window
3. Type `echo $PATH` in Terminal with following result:
```
/opt/local/bin:/opt/local/sbin:/usr/local/Cellar/ruby/1.9.3-p194/bin:/Users/Steven/anaconda/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/Library/TeX/texbin
```
4. Type `which python` in Terminal with following result: `/Users/Steven/anaconda/bin/python`
5. Type `atom` (opens Atom, with all the files open that were open before quitting -- is that normal?)
6. Press Cmd-Shift-T top open terminal pane
7. Type `echo $PATH` and `which python` with the following result:

![atom_screenshot](https://cloud.githubusercontent.com/assets/1272030/8946225/56d6f86a-35c1-11e5-9bf4-42b1fa9fcd72.png)

8. Press Cmd-Opt-I to open the Dev Tools Console
9. Type `process.env.PATH` in the console, and get the following:
```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/Library/TeX/texbin:/opt/local/bin:/opt/local/sbin:/usr/local/Cellar/ruby/1.9.3-p194/bin:/Users/Steven/anaconda/bin
```

Obviously, this is the same as the terminal package is picking up. I'm not sure if you meant the "console" as something different from the Dev Tools Console? 

I'm on OsX Yosemite 10.10.4 if that helps at all.

My init.coffee is empty. I don't know what else would help?

**Comment by bfallik** [comment]
Was there ever a resolution to this issue? I'm having similar trouble launching atom from a shell. The shell environment is not being inherited by atom if I inspect the values in the dev console.

**Comment by lee-dohm** [comment]
Can you reproduce the problem in Safe Mode?

To launch Safe Mode:

1. Completely exit all instances of Atom
1. Launch Atom with the command `atom --safe`

**Comment by bfallik** [comment]
@lee-dohm I did manage to workaround my issue, though I'm not sure if this data will be helpful to anyone else.

First I noticed that some errant Atom processes weren't being exited even when I thought I'd killed everything. This is odd since I'm launching Atom from the terminal, `atom .`, so I'd expect all environment variables to be inherited from the running terminal and not effected by other processes.

However, later on I had to restart the Atom to upgrade and again it failed to find my GOPATH. Again I manually killed some leaked process and restarted and then the env was read correctly.

No, I did not try safe mode but I will try that the next time I encounter this problem.
